include/seastar/core/smp.hh

```diff
@@ -137,6 +137,17 @@ future<smp_service_group> create_smp_service_group(smp_service_group_config ssgc
 /// Note: the function aborts on internal error if smp_service_group is invalid.
 future<> destroy_smp_service_group(smp_service_group ssg) noexcept;
 
+/// Validates an smp_service_group.
+///
+/// Returns true iff the group is valid and was not destroyed.
+///
+/// Note that smp_service_group create and destroy are serialized
+/// under a semaphore and even when this function returns true,
+/// the smp_service_group is not locked and might be destroyed
+/// in parallel. However calling validate_smp_service_group before
+/// attempting to destroy an arbitrary smp_service_group is recommended.
+future<bool> validate_smp_service_group(smp_service_group ssg) noexcept;
+
```

Comment:
What's the point of this? If the service group was destroyed in the past, this returns a random value. It will return false if the smp service group was destroyed and another one with the same id wasn't created in its place, and it will return true if the smp service group was destroyed and another one with the same id was created in its place.

Reply:
that's a valid point.  maybe the service group id should be constructed as `<index, version>` so we can detect stale or invalid id.

Reply:
That's a good idea for debug mode.